server/schedule/region_scatterer.go

```diff
@@ -342,6 +343,27 @@ func (r *RegionScatterer) scatterRegion(region *core.RegionInfo, group string) *
 	op, err := operator.CreateScatterRegionOperator("scatter-region", r.cluster, region, targetPeers, targetLeader)
 	if err != nil {
 		scatterCounter.WithLabelValues("fail", "").Inc()
+
+		if strings.Contains(err.Error(), "cannot build operator for region with nil peer") {
```

Comment:
I think it's better define the errors in `builder` and then use the `err.(type)` like: https://github.com/tikv/pd/blob/master/pkg/apiutil/apiutil.go#L62-L69

Reply:
yeah, I also feel this PR is a bit goofing since A - I tried not to invade `builder`'s logic and B - PingCAP's `errors` pkg is missing some funs, such as `Is` and `As`, so the only thing left is compare the error messages which is not my favor.
I will close this PR and submit a different one using `err.(type)` then...